### PR TITLE
Added exceptions.yaml for v1.29 release

### PR DESCRIPTION
**Pull request by sreeram-venkitesh**
#### What type of PR is this:

/kind documentation

#### What this PR does / why we need it:

This PR adds the `exceptions.yaml` file listing the KEPs which had submitted exception requests post enhancements and code freeze for the v1.29 release

#### Which issue(s) this PR fixes:

No issue has been created. Discussed by Priyanka in the [#release-enhancements](https://kubernetes.slack.com/archives/C02BY55KV7E/p1701248442041249) channel in Slack.

#### Special notes for your reviewer:

Part of the enhancements team's work for v1.29 release cycle.

### CODE REVIEW

File: releases/release-1.29/exceptions.yaml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+- name: "add ProcMount option"
+  issue: 4265
+  date_requested: 2023-10-06
+  date_reviewed: 2023-10-07
+  thread: "https://groups.google.com/g/kubernetes-sig-release/c/emPOBgXxu0Q/m/EFYnxpOxCwAJ"
+  pull_requests:
+    - "https://github.com/kubernetes/enhancements/pull/4266"
+    - "https://github.com/kubernetes/enhancements/pull/4279"
+  status: approved
```
Comment (npolshakova) [inline releases/release-1.29/exceptions.yaml:21]: Let's make this consistent and add quotation marks for string values. We do this in the [previous release exception files](https://github.com/kubernetes/sig-release/blob/86f7b816868ca61caf67f2aea5fbfa9b41af0080/releases/release-1.28/exceptions.yaml#L69). 
```suggestion
  status: "approved"
```

File: releases/release-1.29/exceptions.yaml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+- name: "Optionally Disable HealthCheck Ports for LoadBalancers with ExternalTrafficPolicy=Local"
+  issue: 4144
+  date_requested: 2023-10-10
+  date_reviewed: 2023-10-11
+  thread: "https://groups.google.com/g/kubernetes-sig-network/c/piRehSKuX3Y"
+  pull_requests:
+    - "https://github.com/kubernetes/enhancements/pull/4286"
+  status: denied
```
Comment (npolshakova) [inline releases/release-1.29/exceptions.yaml:39]: Quotation marks for string values here as well

File: releases/release-1.29/exceptions.yaml [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+# Code Freeze Exceptions requested in 1.29
+codeFreeze:
+
+- name: "ClusterTrustBundles"
```
Comment (npolshakova) [inline releases/release-1.29/exceptions.yaml:44]: Do we want to include the old name that they have in the KEP here? [ClusterTrustBundles (previously Trust Anchor Sets)](https://github.com/kubernetes/enhancements/issues/3257)

File: releases/release-1.29/exceptions.yaml [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+  issue: 4008
+  date_requested: 2023-11-01
+  date_reviewed: 2023-11-02
+  thread: "https://groups.google.com/g/kubernetes-sig-release/c/vkHqIH-hE5A/m/FANycyoLEQAJ"
+  pull_requests: 
+    - "https://github.com/kubernetes/kubernetes/pull/121034"
+    - "https://github.com/kubernetes/kubernetes/pull/121461"
+  status: "approved"
+
+- name: "Allow replacement of Pods in a Job when fully terminated"
```
Comment (npolshakova) [inline releases/release-1.29/exceptions.yaml:63]: I think the KEP name is now [Kubelet Support for Image Filesystem being split ](https://github.com/kubernetes/enhancements/issues/4191)

**Review by npolshakova — APPROVED** [review]
LGTM!

**Comment by Priyankasaggu11929** [comment]
/lgtm
/approve

Thank you @sreeram-venkitesh @npolshakova :pray: :heart:

**Pull request merged**